### PR TITLE
Improve grammar, spelling, and punctuation in Extending Wagtail section

### PR DESCRIPTION
**Pull request by albinazs**
I found several typos when studying 'Extending Wagtail' section of the documentation, and decided to check the whole section's text with AI grammar tool. I found more of them, and here is the result.

### CODE REVIEW

**Comment by lb-** [comment]
@albinazs  curious - what was the tool? It seems to be pretty solid.

File: docs/extending/rich_text_internals.md [hunk 1]
```diff
@@ -203,7 +203,7 @@ The editor interface used on rich text fields can be configured with the [WAGTAI
...
 
-Typically, a rich text widget also receives a `features` list, passed from either `RichTextField` / `RichTextBlock` or the `features` option in `WAGTAILADMIN_RICH_TEXT_EDITORS`, which defines the features available in that instance of the editor (see [rich text features](rich_text_features)). To opt in to supporting features, set the attribute `accepts_features = True` on your widget class; the widget constructor will then receive the feature list as a keyword argument `features`.
+Typically, a rich text widget also receives a `features` list, passed from either `RichTextField` / `RichTextBlock` or the `features` option in `WAGTAILADMIN_RICH_TEXT_EDITORS`, which defines the features available in that instance of the editor (see [rich text features](rich_text_features)). To opt into supporting features, set the attribute `accepts_features = True` on your widget class; the widget constructor will then receive the feature list as a keyword argument `features`.
```
Comment (gasman) [inline docs/extending/rich_text_internals.md:206]: There doesn't seem to be any real consensus about whether "opt in to" or "opt into" is correct, but "opt in to" seems to be more common and there are convincing grammatical arguments in favour of it (https://english.stackexchange.com/questions/422848/opt-into-vs-opt-in-to, https://forum.wordreference.com/threads/opt-in-or-opt-into.3560327/) - I think we should keep it.

```suggestion
Typically, a rich text widget also receives a `features` list, passed from either `RichTextField` / `RichTextBlock` or the `features` option in `WAGTAILADMIN_RICH_TEXT_EDITORS`, which defines the features available in that instance of the editor (see [rich text features](rich_text_features)). To opt in to supporting features, set the attribute `accepts_features = True` on your widget class; the widget constructor will then receive the feature list as a keyword argument `features`.
```

**Review by gasman — CHANGES_REQUESTED** [review]
Thanks @albinazs! Most of these changes look good, but there are a few I'm not sure about.

**Comment by albinazs** [comment]
Dear @lb-, I used Grammarly. It highlights possible mistakes in grammar, spelling, and punctuation. After that, I check all suggestions and make a final decision. I can check other docs as well.

Dear @gasman, your knowledge of the context is certainly better than mine:) I accepted all the suggestions - they do make sense. PR is updated accordingly.

**Review by thibaudcolas — APPROVED** [review]
Thank you @albinazs! It seems you’ve addressed everything flagged by @gasman so I’ll merge this now. Some of the changes on commas seem a bit arbitrary to me 😅 but there are plenty of typos too that I’m happy to see fixed.

**Pull request merged**